### PR TITLE
Leaveslips detailview fix

### PR DESCRIPTION
**Pull request by yipingbenjaminliu**
@iamsitting , @ruthnan , @priscilagonzalez , @JoannaRumbley 
please test this for the following:
the rendering of individual leaveslip detail view // should only render selected slip and associated rolls
the rendering of group leaveslip detail view // should only render selected groupslip
usage of personal attendance for trainee and TA // should render everything

### CODE REVIEW

File: ap/attendance/views.py [hunk 1]
```diff
@@ -49,53 +49,69 @@
...
-  events = trainee.events
-  weeks = None
-  disablePeriodSelect = 0
-  if period is not None:
+  groupslips = GroupSlip.objects.none()
+  groupevents = Event.objects.none()
+
+  if leaveslipView:
     weeks = [period * 2, period * 2 + 1]
     start_date = CURRENT_TERM.startdate_of_period(period)
```
Comment (iamsitting) [inline ap/attendance/views.py]: Unused variable? Don't see them anywhere else.

File: ap/leaveslips/views.py [hunk 2]
```diff
@@ -33,7 +33,9 @@ def get_context_data(self, **kwargs):
     ctx = super(LeaveSlipUpdate, self).get_context_data(**kwargs)
     trainee = self.get_object().get_trainee_requester()
     periods = self.get_object().periods
-    ctx.update(react_attendance_context(trainee, period=periods[0], noForm=True))
+    ls_type = kwargs['leaveslip_type']
+    obj_id = kwargs['object_id']
+    ctx.update(react_attendance_context(trainee, leaveslipView=True, leaveslipType=ls_type, period=periods[0], objectId=obj_id))
```
Comment (iamsitting) [inline ap/leaveslips/views.py]: Can we use the request object to reduce the amount of arguments?

**Comment by yipingbenjaminliu** [comment]
@iamsitting , changed

File: ap/attendance/views.py [hunk 3]
```diff
@@ -132,6 +149,7 @@ def react_attendance_context(trainee, period=None, noForm=False):
   return ctx
 
 
+
```
Comment (iamsitting) [inline ap/attendance/views.py]: Extra whitespace

File: ap/attendance/views.py [hunk 4]
```diff
@@ -50,53 +50,70 @@
...
-  weeks = None
-  disablePeriodSelect = 0
-  if period is not None:
+  groupslips = GroupSlip.objects.none()
+  groupevents = Event.objects.none()
+
+  if request:
+    period = request_params['period']
     weeks = [period * 2, period * 2 + 1]
     start_date = CURRENT_TERM.startdate_of_period(period)
```
Comment (iamsitting) [inline ap/attendance/views.py]: Unused variables please remove.

File: ap/attendance/views.py [hunk 5]
```diff
@@ -50,53 +50,70 @@
...
-  # line below is currently replaced with for loop in the if statement
-  # events = trainee.events_in_week_list(weeks) if weeks else trainee.events
-  events = trainee.events
-  weeks = None
-  disablePeriodSelect = 0
-  if period is not None:
+  groupslips = GroupSlip.objects.none()
+  groupevents = Event.objects.none()
+
+  if request:
```
Comment (iamsitting) [inline ap/attendance/views.py]: request not defined.

**Comment by iamsitting** [comment]
👍

**Pull request merged**